src/tests/test_rdymesh.c

```diff
@@ -0,0 +1,114 @@
+// This unit test suite tests the RDyMesh type. See https://cmocka.org for
+// details about the CMocka testing framework.
+
+#include <private/rdymeshimpl.h>
+#include <rdycore.h>
+
+#include "rdycore_tests.h"
+
+// Creates a unit box mesh in 2D.
+// FIXME: Seems like we're already making a lot of assumptions about
+// FIXME: DMs from which we create meshes. Might need some lower-level
+// FIXME: DM creation utilities.
+static PetscErrorCode CreateUnitBoxMesh(PetscInt Nx, PetscInt Ny, DM *dm) {
```

Comment:
How about renaming this function to `Create2DUnitBoxDM`?

Reply:
Good idea!

Reply:
Thanks @knepley . We actually call `DMPlexCreateBoxMesh` around line 22. We have to do some extra things to support the construction of mesh information in `RDyMesh`. It might be worth revisiting what we have in the near future to figure out whether we can streamline it.

Reply:
I also wonder this, and would welcome your input! I started building RDycore based on `ex2.c` in the toy-problems repo, so I'm just getting things in place so we can scrutinize them properly. Less code is fewer moving parts, and fewer stored quantities is a smaller "caching problem", so I'm all in favor of both.

Reply:
Cool. I think we should make a list of the operations we want to use the mesh in. For example, in CEED we really just want iterations over the dofs, and we autoamtically construct these "restrictions" (list of dof numbers in local vectors). We probably also want automated ways of calculating geometric factors. However, I think most of the code doesn't really need to touch the mesh itself.